hibernate-core/src/main/java/org/hibernate/mapping/PersistentClass.java

```diff
@@ -1003,12 +1003,14 @@ protected void checkColumnDuplication(Set<String> distinctColumns, Value value)
 		for ( Selectable columnOrFormula : value.getSelectables() ) {
 			if ( !columnOrFormula.isFormula() ) {
 				Column col = (Column) columnOrFormula;
-				if ( !distinctColumns.add( col.getName() ) ) {
-					throw new MappingException(
-							"Column '" + col.getName()
-									+ "' is duplicated in mapping for entity '" + getEntityName()
-									+ "' (use '@Column(insertable=false, updatable=false)' when mapping multiple properties to the same column)"
-					);
+				if (col.getValue().getColumnUpdateability()[0] || col.getValue().getColumnInsertability()[0]) {
+					if (!distinctColumns.add(col.getName())) {
```

Comment:
I think it would be better to use 

    !ArrayHelper.isAllFalse(col.getValue().getColumnUpdateability())
        || !ArrayHelper.isAllFalse(col.getValue(). getColumnInsertability())

Other than that it looks reasonable.

Reply:
Thank you for the feedback. Applied the changes in the latest commit.